src/App.tsx

```diff
@@ -9,36 +9,116 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import {
+  getUsers, getPosts,
+} from './api/users';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Errors } from './types/Errors';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [currUser, setCurrUser] = useState<User | null>(null);
+  const [postsList, setPostsList] = useState<Post[] | null>(null);
+  const [typeOfError, setTypeOfError] = useState<Errors>(Errors.None);
+  const [isPostsLoading, setPostsLoading] = useState(false);
+  const [currPost, setCurrPost] = useState<Post | null>(null);
+  const [isFormShown, setFormShown] = useState(false);
+
+  async function getUserList() {
+    const usersFromServer = await getUsers();
+
+    setUsers(usersFromServer);
+  }
+
+  const selectUser = (user: User) => {
```

Comment:
Consider renaming 'selectUser' to 'handleUserSelect' to follow React naming conventions for event handlers.